kolibri/core/content/utils/channel_import.py

```diff
@@ -469,13 +469,24 @@ def try_attaching_sqlite_database(self):
                 pass
 
     def try_detaching_sqlite_database(self):
+        from django.db import connection
         # detach the content database from the primary database so we don't get errors trying to attach it again later
         if self.destination.engine.name == "sqlite":
             try:
                 self.destination.session.execute(text("DETACH 'sourcedb'".format(path=self.source_db_path)))
             except OperationalError:
                 # silently ignore if the database was already detached, as then we're good to go
                 pass
+            try:
+                # Importing produces a big fragmentation in the db,
+                # let's vacuum it.
+                self.destination.session.close()
```

Comment:
Let's do this during the `end` method below as the final thing we do, rather than here. That way we have completely finished with our SQLAlchemy connection.